### ISSUE TITLE
Update Dropwizard Metrics to 4.2

### ISSUE THREAD

**Issue by FlorianHockmann** [JanusGraph/janusgraph#3134]
We're currently using version 4.1 of this library for metrics. The update to 4.2 seems to be problematic as it leads to errors due to some problem where a metric is created again with the same name:

```
Caused by: org.janusgraph.diskstorage.PermanentBackendException: Couldn't initialize CQLStoreManager
Caused by: java.lang.IllegalArgumentException: A metric named org.janusgraph.threadpools.CqlStoreManager.executorService.pool.size already exists
```

That is why Dependabot now ignores this minor version (see #3091). So, we have to make the update ourselves.

**Comment by cdegroc** [comment]
An `IllegalArgumentException` is raised when a metric is [registered twice](https://github.com/dropwizard/metrics/blob/v4.1.18/metrics-core/src/main/java/com/codahale/metrics/MetricRegistry.java#L156). This only applies to metrics [added using the register method](https://github.com/dropwizard/metrics/blob/v4.1.18/metrics-core/src/main/java/com/codahale/metrics/MetricRegistry.java#L89).

A number of new metrics were added to the `InstrumentedExecutorService` in dropwizard/metrics 4.2.10.

Before: https://github.com/dropwizard/metrics/blob/v4.1.18/metrics-core/src/main/java/com/codahale/metrics/InstrumentedExecutorService.java#L48-L55
After: https://github.com/dropwizard/metrics/blob/v4.2.10/metrics-core/src/main/java/com/codahale/metrics/InstrumentedExecutorService.java#L51-L87

And `pool.size` is one of them.

Next step is to understand why the metrics are registered twice.

Also, as we already had the `pool.size` metric in janusgraph under the name [queueSize](https://github.com/JanusGraph/janusgraph/blob/v0.6/janusgraph-core/src/main/java/org/janusgraph/diskstorage/configuration/ExecutorServiceInstrumentation.java#L44-L47), we should remove it once we've upgraded to 4.2.10 and make a note of it in the documentation.

**Comment by farodin91** [comment]
Fixed by #3213
